src/templates/TemplateData.ts

```diff
@@ -111,146 +122,114 @@ export class TemplateData {
     try {
-        return <TemplateData>await callWithTelemetryAndErrorHandling('azureFunctions.getTemplateDataFromBackup', reporter, undefined, async function (this: IActionContext): Promise<TemplateData | undefined> {
-            this.suppressErrorDisplay = true;
-            this.properties.isActivationEvent = 'true';
-            const templatesMap: { [runtime: string]: Template[] } = {};
-            const configMap: { [runtime: string]: Config } = {};
-
-            for (const key of Object.keys(ProjectRuntime)) {
-                const runtime: ProjectRuntime = <ProjectRuntime>ProjectRuntime[key];
-                const releaseVersion: string = runtime === ProjectRuntime.one ? v1ReleaseVersion : betaReleaseVersion;
-                if (globalState && globalState.get(`${runtime}-backup`) === releaseVersion) {
-                    const cachedResources: object | undefined = globalState.get<object>(getRuntimeKey(`${resourcesKey}-backup`, runtime));
-                    const cachedTemplates: object[] | undefined = globalState.get<object[]>(getRuntimeKey(`${templatesKey}-backup`, runtime));
-                    const cachedConfig: object | undefined = globalState.get<object>(getRuntimeKey(`${configKey}-backup`, runtime));
-                    if (cachedResources && cachedTemplates && cachedConfig) {
-                        [templatesMap[runtime], configMap[runtime]] = parseTemplates(cachedResources, cachedTemplates, cachedConfig);
-                    }
-                } else {
-                    await downloadAndExtractTemplates(cliFeedJson, releaseVersion);
-                    // only Resources.json has a capital letter
-                    const rawResources: object = <object>await fse.readJSON(path.join(tempPath, 'resources', 'Resources.json'));
-                    const rawTemplates: object[] = <object[]>await fse.readJSON(path.join(tempPath, 'templates', 'templates.json'));
-                    const rawConfig: object = <object>await fse.readJSON(path.join(tempPath, 'bindings', 'bindings.json'));
-
-                    [templatesMap[runtime], configMap[runtime]] = parseTemplates(rawResources, rawTemplates, rawConfig);
-                    verifyTemplatesByRuntime(templatesMap, runtime);
-                    if (globalState) {
-                        globalState.update(`${runtime}-backup`, releaseVersion);
-                        globalState.update(getRuntimeKey(`${templatesKey}-backup`, runtime), rawTemplates);
-                        globalState.update(getRuntimeKey(`${configKey}-backup`, runtime), rawConfig);
-                        globalState.update(getRuntimeKey(`${resourcesKey}-backup`, runtime), rawResources);
-                    }
-                }
-            }
-            return new TemplateData(templatesMap, configMap);
-        });
+        context.properties.templateVersion = templateVersion;
+        context.properties.runtime = runtime;
```

Comment:
Move the 'runtime' telemetry out of this function and into getTemplateData (so that we get it even if we use cached templates)